oneandone/src/main/java/org/apache/jclouds/oneandone/rest/compute/function/ServerToNodeMetadata.java

```diff
@@ -0,0 +1,216 @@
+import org.jclouds.rest.ResourceNotFoundException;
+import org.jclouds.util.InetAddresses2;
+
+public class ServerToNodeMetadata implements Function<Server, NodeMetadata> {
+
+   private final Supplier<Set<? extends Location>> locations;
+   private final Function<Hdd, Volume> fnVolume;
+   private final Function<HardwareFlavour, Hardware> fnHardware;
+   private final OneAndOneApi api;
+
+   private final GroupNamingConvention groupNamingConvention;
+
+   @Inject
+   public ServerToNodeMetadata(Function<Hdd, Volume> fnVolume,
+           Function<HardwareFlavour, Hardware> fnHardware,
+           @Memoized Supplier<Set<? extends Location>> locations,
+           OneAndOneApi api,
+           GroupNamingConvention.Factory groupNamingConvention) {
+      this.locations = locations;
+      this.api = api;
+      this.fnVolume = fnVolume;
+      this.fnHardware = fnHardware;
+      this.groupNamingConvention = groupNamingConvention.createWithoutPrefix();
+   }
+
+   @Override
+   public NodeMetadata apply(final Server server) {
+      checkNotNull(server, "Null server");
+
+      DataCenter dataCenter = api.dataCenterApi().get(server.datacenter().id());
+      Location location = find(locations.get(), idEquals(dataCenter.countryCode().toLowerCase()));
+      Hardware hardware = null;
+      //check if the server was built on a hardware flavour(Fixed instance)
+      if (server.hardware().fixedInstanceSizeId() != null && !"0".equals(server.hardware().fixedInstanceSizeId())) {
+         HardwareFlavour flavour = api.serverApi().getHardwareFlavour(server.hardware().fixedInstanceSizeId());
+         hardware = fnHardware.apply(flavour);
+
+      } else {
+         List<Volume> volumes = Lists.newArrayList();
+         //customer hardware
+         double size = 0d;
+         double minRam = server.hardware().ram();
+         List<Hdd> hdds = server.hardware().hdds();
+
+         if (server.hardware().hdds().isEmpty()) {
+            hdds = api.serverApi().getHardware(server.id()).hdds();
+         }
+
+         size = getHddSize(hdds);
+         volumes = convertHddToVolume(hdds);
+
+         if (minRam < 1) {
+            minRam = 512;
+         } else {
+            minRam = minRam * 1024;
+         }
+
+         List<Processor> processors = new ArrayList<Processor>();
+         for (int i = 0; i < server.hardware().vcore(); i++) {
+            Processor proc = new Processor(server.hardware().coresPerProcessor(), 1d);
+            processors.add(proc);
+         }
+         AutomaticHardwareIdSpec id = AutomaticHardwareIdSpec.automaticHardwareIdSpecBuilder(server.hardware().vcore(), (int) minRam, Optional.of((float) size));
+         hardware = new HardwareBuilder()
+                 .ids(id.toString())
+                 .ram((int) minRam)
+                 .processors(ImmutableList.copyOf(processors))
+                 .hypervisor("kvm")
+                 .volumes(volumes)
+                 .location(location)
+                 .build();
```

Comment:
I don't think so. That is there because Google has its own way of deploying custom hardware profiles, encoding the values in the "size uri".
Let's leave this as-is in this PR. It's not a critical thing to change.